### PR TITLE
feat: change blobclient from management api to rest api

### PR DESCRIPTION
**Pull request by cvvz**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
feat: change blobclient from management api to rest api

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
feat: change blobclient from management api to rest api
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
feat: change blobclient from management api to rest api
```

### CODE REVIEW

**Comment by MartinForReal** [comment]
Could you please update mock function and squash commit? Thanks!

**Comment by cvvz** [comment]
> Could you please update mock function and squash commit? Thanks!
Mock updated, and I will refine the unit test for blobclient.

File: pkg/azureclients/blobclient/interface.go [hunk 1]
```diff
@@ -20,12 +20,22 @@ import (
...
 
 	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2021-09-01/storage"
+	"sigs.k8s.io/cloud-provider-azure/pkg/retry"
+)
+
+const (
+	// APIVersion is the API version for storage.
+	APIVersion = "2021-09-01"
+	// AzureStackCloudAPIVersion is the API version for Azure Stack
+	AzureStackCloudAPIVersion = "2019-03-01"
```
Comment (andyzhangx) [inline pkg/azureclients/blobclient/interface.go:30]: do you have the supported api version list for azure stack? Thanks.  @JesusAlvarezTorres do you have a link?

Comment (andyzhangx) [reply pkg/azureclients/blobclient/interface.go:30]: let's use `2019-06-01` per and also add this link in the comments:
https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-profiles-azure-resource-manager-versions?view=azs-2206

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
/retest

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by monianshouhou** [comment]
There exist some unstable factors in basic-lb test. Please use following instruction to rerun the test:
/azp run kubernetes-sigs.cloud-provider-azure.basic_lb

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**